File: solutions/beecrowd/1873/1873.cpp
#include <cstdint>
#include <iostream>
#include <string>

std::string rock_paper_scissors_lizard_spock_judge(
    std::string rajeshs_hand, std::string sheldons_hand) {
    if (sheldons_hand == rajeshs_hand) {
        return "empate";
    }
    if ((rajeshs_hand == "tesoura"
         && (sheldons_hand == "papel" || sheldons_hand == "lagarto"))
        || (rajeshs_hand == "papel"
            && (sheldons_hand == "pedra" || sheldons_hand == "spock"))
        || (rajeshs_hand == "pedra"
            && (sheldons_hand == "lagarto" || sheldons_hand == "tesoura"))
        || (rajeshs_hand == "lagarto"
            && (sheldons_hand == "spock" || sheldons_hand == "papel"))
        || (rajeshs_hand == "spock"
            && (sheldons_hand == "tesoura" || sheldons_hand == "pedra"))) {
        return "rajesh";
    }
    return "sheldon";
}

int main() {
    std::int16_t c;
    std::string sheldons_hand, rajeshs_hand;

    std::cin >> c;

    while (c--) {
        std::cin >> rajeshs_hand >> sheldons_hand;
        std::cout << rock_paper_scissors_lizard_spock_judge(
            rajeshs_hand, sheldons_hand)
                  << std::endl;
    }

    return 0;
}

File: solutions/beecrowd/1873/1873.py
import sys

input = sys.stdin.readline
output = sys.stdout.write


def rock_paper_scissors_lizard_spock_judge(rajeshs_hand, sheldons_hand):
    if rajeshs_hand == sheldons_hand:
        return 'empate'
    if rajeshs_hand == 'tesoura' and sheldons_hand in ('papel', 'lagarto'):
        return 'rajesh'
    if rajeshs_hand == 'papel' and sheldons_hand in ('pedra', 'spock'):
        return 'rajesh'
    if rajeshs_hand == 'pedra' and sheldons_hand in ('lagarto', 'tesoura'):
        return 'rajesh'
    if rajeshs_hand == 'lagarto' and sheldons_hand in ('spock', 'papel'):
        return 'rajesh'
    if rajeshs_hand == 'spock' and sheldons_hand in ('tesoura', 'pedra'):
        return 'rajesh'
    return 'sheldon'


c = int(input())

for _ in range(c):
    rajeshs_hand, sheldons_hand = input().split()
    output(
        f'{rock_paper_scissors_lizard_spock_judge(rajeshs_hand, sheldons_hand)}\n'
    )
